src/main/kotlin/org/rust/ide/annotator/RsExpressionAnnotator.kt

```diff
@@ -79,10 +74,17 @@ private class RedundantParenthesisVisitor(private val holder: AnnotationHolder)
     override fun visitParenExpr(o: RsParenExpr) =
         o.expr.warnIfParens("Redundant parentheses in expression")
 
+    override fun visitLetDecl(o: RsLetDecl) {
+        o.expr.warnIfParens("Redundant parentheses in let declaration")
+    }
+
     private fun RsExpr?.warnIfParens(message: String) {
         if (this is RsParenExpr) {
-            holder.createWeakWarningAnnotation(this, message)
-                .registerFix(RemoveParenthesesFromExprIntention())
+            if ((this.parentOfType<RsCondition>() != null || this.parentOfType<RsMatchExpr>() != null)
+                && this.children.singleOrNull() is RsStructLiteral) {
+            } else
```

Comment:
That's one puzzling `if`, with an empty `then` branch, and multiline `else` branch without braces :) 

I wonder if we can simplify this a bit, and remove duplication with the intention as well? Something like this should work, perhaps? 

```Kotlin
class RemoveParenthesesFromExprIntention {
  fun isAvailable(expr: RsParanExpr): Boolean { /* check that we can safely remove parenthesis */  }
  override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): RsParenExpr? {
    val parenExpr = element.parentOfType<RsParenExpr>()
    return if (isAvailable(parenExpr))  parenExpr else null
  }
}

// In the annotator
 private fun RsExpr?.warnIfParens(message: String) {
    if (this !is RsParenExpr) return
    val fix = RemoveParenthesesFromExprIntention()
    if (fix.isAvailable(this)) holder.createWeakWarningAnnotation(...)
}

```